packages/ra-core/src/form/FormField.tsx

```diff
@@ -15,32 +13,34 @@ export const isRequired = validate => {
     return false;
 };
 
-interface Props {
-    component: ComponentType<InputProps>;
-    defaultValue: any;
+interface Props extends Omit<FieldProps<any, HTMLElement>, 'validate'> {
+    defaultValue?: any;
     input?: any;
     source: string;
-    validate: Validator | Validator[];
+    validate?: Validator | Validator[];
 }
 
-export const FormFieldView: SFC<Props> = ({ input, ...props }) =>
-    input ? ( // An ancestor is already decorated by Field
+export const FormField: SFC<Props> = ({ input, validate, ...props }) => {
+    const sanitizedValidate = Array.isArray(validate)
+        ? composeValidators(validate)
+        : validate;
+
+    return input ? ( // An ancestor is already decorated by Field
```

Comment:
contrary to redux-form, I don't think react-final-form injects an `input` prop, so you'll need another trick to avoid double decoration

Reply:
Nope, it does: https://github.com/final-form/react-final-form#fieldrenderprops